### ISSUE TITLE
Latest FAWE does not work on Java 9+ sun.misc.SharedSecrets

### ISSUE THREAD

**Issue by Foorack** [boy0001/FastAsyncWorldedit#870]
# Bug report

**Debug paste link**:

```[16:07:58] [Server thread/ERROR]: Could not pass event ChunkLoadEvent to FastAsyncWorldEdit v18.01.10-274bc6d-962-18.1.3
java.lang.NoClassDefFoundError: sun/misc/SharedSecrets
        at com.boydti.fawe.bukkit.v0.ChunkListener.onChunkLoad(ChunkListener.java:205) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor358.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[paper-1.12.1.jar:git-Paper-1293]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:78) ~[paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:515) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:500) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.Chunk.loadNearby(Chunk.java:995) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:54) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOProvider.callStage2(ChunkIOProvider.java:16) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor$Task.finish(AsynchronousExecutor.java:188) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.util.AsynchronousExecutor.finishActive(AsynchronousExecutor.java:347) [paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.chunkio.ChunkIOExecutor.tick(ChunkIOExecutor.java:36) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:859) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [paper-1.12.1.jar:git-Paper-1293]
        at java.lang.Thread.run(Thread.java:844) [?:?]
```

**Description of the problem:**
Broken by this commit https://github.com/boy0001/FastAsyncWorldedit/commit/e1e31beff09796670773a89a4468c5020df9577d#diff-c5b74d6eaa5d5886401ab2715bfdf615

**How to replicate**:
Run version v18.01.10-27fbc6d-962-18.1.3 or later on Java 9 or up.

**Checklist**:

- [X] I made sure there aren't duplicates of this report [(Use Search)](https://github.com/boy0001/FastAsyncWorldedit/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [X] I made sure I am using an up-to-date version of FAWE
- [X] I Made sure the bug/error is not caused by any other plugin

**Comment by krusic22** [comment]
Can confirm. Also broke on Java 10 :D

**Comment by Foorack** [comment]
@krusic22 "Java 10" Like living on the edge huh? :sweat_smile:
Thanks for the comment, I updated the issue to reflect that.

**Comment by Panossa** [comment]
To quote someone from the Discord who already got it to work on Java 9: 
"I found the problem, FAWE doesn't need Java EE but you need to add `--add-modules java.xml.bind` to the command line for the plugin to work in Java 9."

**Comment by Foorack** [comment]
@Panossa Yes, I know. That was me who said that, a while ago. This is a different issue. The old issue was that FAWE couldn't find the `java.xml.bind` package. This issue is about @boy0001 **USING CODE WHICH HAS BEEN DEPRECATED SINCE 1996**

https://user-images.githubusercontent.com/5008081/27436549-e3e33e1e-575f-11e7-9c03-bbf4f054d0b0.png

**Comment by Panossa** [comment]
Lol, my bad xD
Btw, took them long enough to remove deprecated stuff xD

**Comment by boy0001** [comment]
Because spigot doesn't provide a way to tell what caused the physics event, so I need to analyze the stacktrace, which has a high performance cost if it's done normally. Anywho, I'm now using reflection to access the native methods, so it should work on java 9. 

@Panossa 
It wasn't removed, just moved.

**Comment by Foorack** [comment]
FAWE still doesn't work on Java 9. :frowning: 
Should I report this as a separate issue? Posting it here as I see `ChunkListener_9` in the StackTrace.
Also do you have to use reflection? Wouldn't it be possible to solve with the StackWalker? :thinking: 

```[09:03:53 INFO]: (FAWE) Plugin 'WorldGuard' found. Using it now.
[09:03:53 WARN]: [FastAsyncWorldEdit] Task #20 for FastAsyncWorldEdit v18.01.11-4b7c1e9-964-18.1.5 generated an exception
org.bukkit.plugin.IllegalPluginAccessException: Unable to find handler list for event org.bukkit.event.block.BlockEvent. Static getHandlerList method required!
        at org.bukkit.plugin.SimplePluginManager.getRegistrationClass(SimplePluginManager.java:606) ~[paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.plugin.SimplePluginManager.getRegistrationClass(SimplePluginManager.java:604) ~[paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.plugin.SimplePluginManager.registerEvents(SimplePluginManager.java:547) ~[paper-1.12.1.jar:git-Paper-1293]
        at com.boydti.fawe.bukkit.v0.ChunkListener.<init>(ChunkListener.java:33) ~[?:?]
        at com.boydti.fawe.bukkit.v0.ChunkListener_9.<init>(ChunkListener_9.java:39) ~[?:?]
        at com.boydti.fawe.bukkit.FaweBukkit$1.run(FaweBukkit.java:156) ~[?:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:58) ~[paper-1.12.1.jar:git-Paper-1293]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:356) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:833) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [paper-1.12.1.jar:git-Paper-1293]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [paper-1.12.1.jar:git-Paper-1293]
        at java.lang.Thread.run(Thread.java:844) [?:?]
```

**Comment by boy0001** [comment]
@Foorack 
Sorry, the above error was with trying to register the wrong event. It's not related to java 9, and has since been fixed. 

I've also scrapped using reflection, it didn't work very well. StackWalker would do the same thing, but be much slower. I found that abusing FAWE in java 9 lead to a much quicker StackOverflow (i.e. the server crashes). So I needed to change the class to detect the abuse quicker, which wasn't possible by analyzing the stack without incurring a significant performance penalty.

I've settled for something that performs better, has quicker detection of abuse, but may introduce some false positives.
